### DISCUSSION TITLE
Using mblaze with wezterm

### DISCUSSION THREAD

**Discussion by xmailla** [wezterm/wezterm#6517] [Q&A]
### What Operating System(s) are you running on?

macOS

### WezTerm version

wezterm 20240203-110809-5046fc22

### Ask your question!

Hello,

I use the Mblaze suite of tools to manage my emails (read, filter, reply, etc.).
I try to use and adapt a tool made by the community that allows to have a sort of "console" and from which, we spawn commands via tmux.
I do not use tmux ;) So I want to be able to modify it and adapt it to my needs and to spawn wezterm clients.

One of the commands used is: 
`tmux new-window -n "$title" "mless $*"`

So I first modified it so that it does:
`wezterm cli spawn set-tab-title $title -- mless $*`

the mless command is in my PATH:
`/usr/local/bin/mless`

The problem is that when I type 6580 (the number of the email I want to read) from my tool, I get this error:

```
Unable to spawn set-tab-title because:
No viable candidates found in PATH "/Users/x/perl5/perlbrew/bin:/Users/x/perl5/perlbrew/perls/perl-5.32.0/bin:/Users/x/.local/share/zinit/polaris/bin:/Users /x/.local/bin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.crypt exd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/opt/pkg/sbin:/opt/pkg/bin:/Library/Apple/usr/bin"
⚠️ Process "set-tab-title '[Reception' '#6580]' -- mless 6580" in domain "local" didn't exit cleanly
Exited with code 1.
This message is shown because exit_behavior="CloseOnCleanExit"
```

So I have 2 questions:

1.  how to process this message and how to make it display
2. how to "exit" this error screen and return to the previous screen

Thank you for your help.
(message translated into English via google translate)

**Comment by wez** [comment]
`wezterm cli spawn set-tab-title $title -- mless $*` tells wezterm to spawn `set-tab-title $title -- mless $*` in a new window.

The error message you shared shows that there is no command named `set-tab-title` present in the `$PATH` that wezterm has inherited from your environment.

You can resolve this issue either by changing `set-tab-title` to be the full path to that program, or removing it and just spawning your `mless` command:

`wezterm cli spawn /some/path/to/set-tab-title $title -- /usr/local/bin/mless $*`

or:

`wezterm cli spawn /usr/local/bin/mless $*`

See also: https://wezfurlong.org/wezterm/faq.html#im-on-macos-and-wezterm-cannot-find-things-in-my-path

**Comment by wez** [comment]
> 2\. how to "exit" this error screen and return to the previous screen

You can close the window/pane in the usual way, either by clicking on the close button in the window title, or by triggering a close action such as `CloseCurrentPane` or `CloseCurrentTab`.  https://wezfurlong.org/wezterm/config/default-keys.html shows the default assignments for various keys.
